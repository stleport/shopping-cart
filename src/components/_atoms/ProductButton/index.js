import React from 'react';
import PropTypes from 'prop-types';
import {
  ReactComponent as Plus
} from '../../../assets/images/plus.svg';
import {
  ReactComponent as Minus
} from '../../../assets/images/minus.svg';

const ProductButton = ({
  iconName,
  iconColor,
  bgColor,
  enabled,
  onClick
}) => {
  const classNames = function classNames() {
    const bgColorModifier = bgColor === 'dark' ? '--dark' : '--light';
    const enabledAttribute = enabled || ' disabled';
    const button = `st-Edit-button__button st-Edit-button__button${bgColorModifier} button${enabledAttribute}`;
    const icon = `st-Edit-button__icon--${iconColor}`;
    return {
      button,
      icon
    };
  };

  return ( 
    <React.Fragment>
      <button
        className={classNames().button}
        type="button"
        onClick={onClick}
      >
        {iconName === 'plus' ? (
          <Plus 
            fill={iconColor}
            className={classNames().icon}
          />
        ) : (
          <Minus 
            fill={iconColor}
            className={classNames().icon}
          />
        )}
      </button>
    </React.Fragment>
  );
};

ProductButton.defaultProps = {
  enabled: true
};

ProductButton.propTypes = {
  iconName: PropTypes.string.isRequired,
  iconColor: PropTypes.string.isRequired,
  bgColor: PropTypes.string.isRequired,
  enabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

export default ProductButton;