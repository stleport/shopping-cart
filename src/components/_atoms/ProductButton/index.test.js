import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ProductButton from '.';

function ProductButtonSetup(newProps = {}) {
  const props = {
    iconName: 'plus',
    iconAlign: 'right', 
    iconFloat: 'floated',
    iconColor: 'dark',
    enabled: true,
    bgColor: 'dark',
    onClick: jest.fn()
  };

  const ProductButtonJSX = (
    <ProductButton {...{ ...props, ...newProps }} />
  );

  const ProductButtonComponent = mount(ProductButtonJSX);
  return {
    ProductButtonComponent,
    ProductButtonJSX,
    props
  };
}

describe('ProductButton', () => {

  it('renders correctly ProductButton component', () => {
    const { ProductButtonJSX } = ProductButtonSetup();
    const ProductButtonComponent = renderer.create(ProductButtonJSX).toJSON();
    expect(ProductButtonComponent).toMatchSnapshot();
  });

  it('click handler is called onClick', () => {
    const { ProductButtonComponent, props } = ProductButtonSetup();
    ProductButtonComponent.find('button').simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });
  it('button should not be disabled when enabled prop is true', () => {
    const { ProductButtonComponent } = ProductButtonSetup();
    expect(ProductButtonComponent.find('button').hasClass('disabled')).toBe(false);
  });
  it('button should be disabled when enabled prop is false', () => {
    const { ProductButtonComponent } = ProductButtonSetup({ enabled: false });
    expect(ProductButtonComponent.find('button').hasClass('disabled')).toBe(true);
  });
});
