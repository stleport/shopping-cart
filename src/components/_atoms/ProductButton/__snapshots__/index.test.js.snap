// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductButton renders correctly ProductButton component 1`] = `
<button
  className="st-Edit-button__button st-Edit-button__button--dark buttontrue"
  onClick={[MockFunction]}
  type="button"
>
  <svg
    className="st-Edit-button__icon--dark"
    fill="dark"
  >
    plus.svg
  </svg>
</button>
`;
